security/advancedtls/advancedtls.go

```diff
@@ -81,6 +81,11 @@ type GetRootCAsResults struct {
 // RootCertificateOptions contains a field and a function for obtaining root
 // trust certificates.
 // It is used by both ClientOptions and ServerOptions.
+// If users want to use default verification, but did not provide a valid
```

Comment:
It might be a little bit redundant to document your logic here, since it could be easily observed from the code.
Can we rephrase it as "If users want to use default verification, but did not provide a valid RootCertificateOptions, we will use the system default trust certificates"?